crates/oxc_linter/src/lib.rs

```diff
@@ -81,19 +110,38 @@ impl Linter {
     }
 
     pub fn run<'a>(&self, ctx: LintContext<'a>) -> Vec<Message<'a>> {
+        let mut rule_timings = self.rule_timings.lock().unwrap();
```

Comment:
I might need some help with the concurrency story here @Boshen, this obviously causes things to be _very_ slow since we're obtaining a lock too frequently